include/alpaka/block/shared/st/BlockSharedMemStMember.hpp

```diff
@@ -0,0 +1,146 @@
+/* Copyright 2020 Jeffrey Kelling
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/block/shared/st/Traits.hpp>
+#include <alpaka/core/Assert.hpp>
+
+#include <type_traits>
+#include <cstdint>
+
+namespace alpaka
+{
+    namespace block
+    {
+        namespace shared
+        {
+            namespace st
+            {
+                namespace detail
+                {
+                    //#############################################################################
+                    //! Implementation of static block shared memory provider.
+                    template<unsigned int TDataAlignBytes = 4>
+                    class BlockSharedMemStMemberImpl
+                    {
+                        mutable unsigned int m_allocdBytes = 0;
+                        mutable uint8_t* m_mem;
+#ifndef NDEBUG
+                        const unsigned int m_capacity;
+#endif
+
+                        template<typename T>
+                        static constexpr size_t alignPitch()
+                        {
+                            return (sizeof(T)/TDataAlignBytes + (sizeof(T)%TDataAlignBytes>0))*TDataAlignBytes;
+                        }
```

Comment:
Could you please move the private members down? All other classes in alpaka have it this way.